sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala

```diff
@@ -52,50 +52,36 @@ object ScalaNativePluginInternal {
     taskKey[Unit]("Warn if JVM 7 or older is used.")
 
     lazy val scalaNativeDependencySettings: Seq[Setting[_]] = {
-      val organization = "org.scala-native"
       val nativeStandardLibraries =
         Seq("nativelib", "clib", "posixlib", "windowslib", "javalib", "auxlib")
 
       Seq(
-        libraryDependencies ++= Seq(
-          organization %%% "test-interface" % nativeVersion % Test
-        ),
-        libraryDependencies += CrossVersion
-          .partialVersion(scalaVersion.value)
-          .fold(throw new RuntimeException("Unsupported Scala Version")) {
-            case (2, _) =>
-              organization %%% "scalalib" % scalalibVersion(
-                scalaVersion.value,
-                nativeVersion
-              )
-            case (3, _) =>
-              organization %%% "scala3lib" % scalalibVersion(
-                scalaVersion.value,
-                nativeVersion
-              )
-          },
-        libraryDependencies ++= nativeStandardLibraries.map(
-          organization %%% _ % nativeVersion
-        ),
+        libraryDependencies ++= {
+          val org = nativeOrgName
+          val ver = nativeVersion
+          val scalalib = CrossVersion.partialVersion(scalaVersion.value) match {
+            case Some((2, _)) => "scalalib"
+            case Some((3, _)) => "scala3lib"
+            case _ => throw new RuntimeException("Unsupported Scala Version")
+          }
+          Seq(
+            org %%% "test-interface" % ver % Test,
+            org %%% scalalib % scalalibVersion(scalaVersion.value, ver),
+            compilerPlugin(org % "nscplugin" % ver cross CrossVersion.full)
+          ) ++ nativeStandardLibraries.map(org %%% _ % ver)
+        },
         excludeDependencies ++= {
           // Exclude cross published version dependencies leading to conflicts in Scala 3 vs 2.13
           // When using Scala 3 exclude Scala 2.13 standard native libraries,
           // when using Scala 2.13 exclude Scala 3 standard native libraries
           // Use full name, Maven style published artifacts cannot use artifact/cross version for exclusion rules
+          val suffix = ScalaNativeCrossVersion.scalaNativePart +
+            (if (scalaVersion.value.startsWith("3.")) "_2.13" else "_3")
+          val exclRule = ExclusionRule(nativeOrgName)
           nativeStandardLibraries.map { lib =>
-            val scalaBinVersion =
-              if (scalaVersion.value.startsWith("3.")) "2.13"
-              else "3"
-            ExclusionRule()
-              .withOrganization(organization)
-              .withName(
-                s"${lib}_native${ScalaNativeCrossVersion.currentBinaryVersion}_${scalaBinVersion}"
-              )
+            exclRule.withName(s"${lib}_${suffix}")
```

Comment:
If possible let's split the suffix into Scala Native and Scala binary parts, when looking at interpolated string it's clear that we're looking on platform specific library

Reply:
Also, when looking at it it seems that we unnecessary exclude Scala 2.13 dependencies in Scala 2.12 - it would never be needed. If you want you an fix it as well in here. So potentially match on CrossVersion.partialVersion would allow to cleanly split these things up.